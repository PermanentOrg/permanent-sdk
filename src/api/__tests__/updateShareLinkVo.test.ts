import nock from 'nock';
import { updateShareLinkVo } from '..';

describe('updateShareLinkVo', () => {
  test('should update share link', async () => {
    nock('https://permanent.local')
      .post(
        '/api/share/updateShareLink',
        {
          shareByUrlId: 1,
          maxUses: 10,
          previewToggle: true,
          autoApproveToggle: true,
          defaultAccessRole: 'access.role.editor',
          expiresDT: '2022-01-01T00:00:00Z',
        },
      )
      .replyWithFile(
        200,
        `${__dirname}/fixtures/updateShareLinkVo/shareLinkVo.json`,
        { 'Content-Type': 'application/json' },
      );

    const shareLinkVo = await updateShareLinkVo(
      {
        bearerToken: '12345',
        baseUrl: 'https://permanent.local/api',
      },
      {
        shareByUrlId: 1,
        maxUses: 10,
        previewToggle: true,
        autoApproveToggle: true,
        defaultAccessRole: 'access.role.editor',
        expiresDT: '2022-01-01T00:00:00Z',
      },
    );

    expect(shareLinkVo).toMatchSnapshot();
  });
});
