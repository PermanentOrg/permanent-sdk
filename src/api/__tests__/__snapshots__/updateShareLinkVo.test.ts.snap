// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`updateShareLinkVo should update share link 1`] = `
{
  "AccountVO": null,
  "ArchiveVO": null,
  "FolderVO": null,
  "RecordVO": null,
  "ShareVO": null,
  "autoApproveToggle": 0,
  "byAccountId": 1770,
  "byArchiveId": 1765,
  "createdDT": "2024-02-02T19:05:25",
  "defaultAccessRole": "access.role.editor",
  "expiresDT": null,
  "folder_linkId": 8306,
  "maxUses": 0,
  "previewToggle": 0,
  "shareUrl": "https://local.permanent.org/share/4fe0ce05ecd16d64d7fd223e66e279ea8d692d300c4ae61b0a5ed39530ff87fc",
  "shareby_urlId": 203,
  "status": "status.generic.ok",
  "updatedDT": "2024-02-02T19:05:53",
  "urlToken": "4fe0ce05ecd16d64d7fd223e66e279ea8d692d300c4ae61b0a5ed39530ff87fc",
  "uses": 0,
}
`;
